tendrl/commons/flows/import_cluster/__init__.py

```diff
@@ -15,6 +16,30 @@
 
 
 class ImportCluster(flows.BaseFlow):
+    def _probe_and_mark_provisioner(self, node_list, integration_id):
+        for node in node_list:
+            try:
+                tags = json.loads(
+                    NS._int.client.read(
+                        "nodes/%s/NodeContext/tags" % node
+                    ).value
+                )
+                if "provisioner/gluster" in tags and \
+                    "provisioner/%s" % integration_id not in tags:
+                    tags.append("provisioner/%s" % integration_id)
+                    NS._int.wclient.write(
```

Comment:
NodeContext.tags is a "list" type in Tendrl definitions, which means it is supposed to be a json array. Hence this is going to fail.

Create an actual NS.tendrl.objects.NodeContext(node_id=node) and add the new tag to its "tags" attribute and call .save()